contracts/GoodGhostingPolygonCurve.sol

```diff
@@ -0,0 +1,674 @@
+pragma solidity 0.6.11;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Pausable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./curve/ICurvePool.sol";
+import "./curve/ICurveGauge.sol";
+
+contract GoodGhostingPolygonCurve is Ownable, Pausable {
+    using SafeMath for uint256;
+
+    /// @notice curve rewards per player
+    uint256 public curveRewardsPerPlayer;
+    /// @notice wmatic rewards per player
+    uint256 public rewardsPerPlayer;
+    /// @notice Stores the total amount of net interest received in the game.
+    uint256 public totalGameInterest;
+    /// @notice total principal amount
+    uint256 public totalGamePrincipal;
+    /// @notice performance fee amount allocated to the admin
+    uint256 public adminFeeAmount;
+    /// @notice total amount of incentive tokens to be distributed among winners
+    uint256 public totalIncentiveAmount = 0;
+    /// @notice Controls the amount of active players in the game (ignores players that early withdraw)
+    uint256 public activePlayersCount = 0;
+    /// @notice The amount to be paid on each segment
+    uint256 public immutable segmentPayment;
+    /// @notice The number of segments in the game (segment count)
+    uint256 public immutable lastSegment;
+    /// @notice When the game started (deployed timestamp)
+    uint256 public immutable firstSegmentStart;
+    /// @notice The time duration (in seconds) of each segment
+    uint256 public immutable segmentLength;
+    /// @notice The early withdrawal fee (percentage)
+    uint256 public immutable earlyWithdrawalFee;
+    /// @notice The performance admin fee (percentage)
+    uint256 public immutable customFee;
```

Comment:
we can probably use smaller data types here to and pack them with the ones below